### PR TITLE
Angus/remote file requests

### PR DESCRIPTION
**Pull request by veggiesaurus**
**Description**

WIP for loading images from hips2fits server

**Checklist**

For linked issues (if there are):
- [x] ~assignee and label added~
- [x] ~ZenHub issue connection, board status, and estimate updated~

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] changelog updated / no changelog update needed
- [x] unit test added (for functions with no dependenies)
- [x] API documentation added (for public variables and methods in stores)

For dependencies:
- [x] e2e test passing / corresponding fix added / new e2e test created
- [x] protobuf version bumped / no protobuf version bumped needed
- [x] protobuf updated to the latest dev commit / no protobuf update needed
- [ ] corresponding ICD test fix added (`BackendService` changed) / ~no ICD test fix needed (`BackendService` unchanged)~
- [ ] user manual prepared (for large new features)

### CODE REVIEW

**Comment by YuHsuan-Hwang** [comment]
Ready for review.

Changes:
* Changed the "Online catalog query" dialog to "Online data query" dialog, and added two tabs for catalog queries and HiPS survey queries.
* Enabled the dialog button and the menu option when there is no image loaded.
* Added HiPS survey query UI. Users can query by either the object name or the center coordinates.
* Pinned the action buttons at the button of the dialog. The "Query" button is disabled if any of the input is not valid or a query is in progress. The "Clear" button resets the inputs.
* When a query is sent, a spinner is shown and all UI is disabled.
* Added documentation and unit tests for `HipsQueryStore`.

This PR is currently for the basic features of HiPS data queries. There are some possible enhancements, and I would suggest we file a new issue for them:
* ~For the HiPS survey text input, search for available surveys and show them in a dropdown.~
* For center coordinates and fov, add shortcut buttons to set by the parameters of an active image (similar UI in the online catalog query widget).
* Add cancel button for the query (is this possible?).
* (optional) Adjust the catalog query UI for consistency.

Companion PRs: https://github.com/CARTAvis/carta-backend/pull/1359, https://github.com/CARTAvis/carta-protobuf/pull/93

**Comment by YuHsuan-Hwang** [comment]
Updates:
* Added a link to the list of the surveys.
* Fixed failure of requesting images in galactic coordinates by fixing the typo.
* Fixed errors when resuming sessions and saving workspaces. The requested images are labeled as in-memory images and will not be resumed when resuming sessions and saving workspaces.

**Comment by YuHsuan-Hwang** [comment]
Update: for the HiPS survey text input, search for available surveys and show them in a dropdown.
* Only surveys with fits format available are listed.
* If fetching the list failed, the dropdown will not be shown. Users are allowed to input arbitrary survey names.

**Comment by kswang1029** [comment]
@YuHsuan-Hwang Let's have the following minor improvements before concluding this PR. These are mainly for improvement the UX and reduce user errors.

1) I suggest we move the "Dimension (px)" entries to be above the "Field of view (deg)" entries as they belong to the same purpose.
2) Rename "Center (deg)" to "ICRS center (deg)". This is to reduce user error by interpreting the "Coordinate system" entries as the input coordinate system for the center. 
3) Rename "Coordinate system" to "Output system" to make it clearer that the selected system is for the output FITS image. 

Screenshot 2024-07-01 at 11 26 19

After these minor changes, this PR is ok to go. We can leave the following improvements as another PR in the future:

- having the ability to refer to the FOV center in the image viewer as the ICRS Center coordinate. This would greatly improve the UX as finding out ICRS coordinate in "deg" is not that trivial. Usually we use sexagesimal format for ICRS and decimal deg for galactic.

**Comment by YuHsuan-Hwang** [comment]
@kswang1029 Thanks for the comments. The UI design is adjusted.

**Review by kswang1029 — CHANGES_REQUESTED** [review]
@YuHsuan-Hwang found a regression: when we load an image and launch the online data query dialog, the center coordinate is not set automatically and appears as "<bad>". This is causing SIMBAD and VizieR e2e tests failing.

Screenshot 2024-07-10 at 10 26 13

**Review by kswang1029 — APPROVED** [review]
looks good. e2e tests are adjusted accordingly and a new e2e test is added. 👍

**Comment by loveluthien** [comment]
@YuHsuan-Hwang The file browser is open when launching CARTA. Is it better if the file browser can be closed automatically at some point when loading a HiPS file?

**Comment by YuHsuan-Hwang** [comment]
> @YuHsuan-Hwang The file browser is open when launching CARTA. Is it better if the file browser can be closed automatically at some point when loading a HiPS file?

The file browser could also be opened by the user. Not sure if we should close it anyway.

File: src/stores/CatalogOnlineQuery/CatalogOnlineQueryConfigStore.ts [hunk 1]
```diff
@@ -54,6 +54,8 @@ export class CatalogOnlineQueryConfigStore {
         this.vizierResource = new Map();
         this.vizierKeyWords = "";
 
+        this.resetSearchRadius();
+
```
Comment (loveluthien) [inline src/stores/CatalogOnlineQuery/CatalogOnlineQueryConfigStore.ts:58]: Once `this.resetSearchRadius()` is set, does the below reaction section become redundent?

Comment (YuHsuan-Hwang) [reply src/stores/CatalogOnlineQuery/CatalogOnlineQueryConfigStore.ts:58]: The reaction is required for reseting the inputs again when the active frame is changed.

**Review by loveluthien — APPROVED** [review]
Nice work.

**Review by YuHsuan-Hwang — APPROVED** [review]
Awaiting for backend PR https://github.com/CARTAvis/carta-backend/pull/1359 solving AppImage related issue.
@acdo2002 Might also need ICD test update since `BackendService.ts` is changed.

**Pull request merged**